mmv1/products/edgecontainer/Cluster.yaml

```diff
@@ -0,0 +1,492 @@
+        properties:
+          - !ruby/object:Api::Type::Boolean
+            name: "disabled"
+            default_from_api: true
+            description: |
+              Whether Ingress is disabled.
+          - !ruby/object:Api::Type::String
+            name: "ipv4Vip"
+            default_from_api: true
+            description: |
+              Ingress VIP.
+  - !ruby/object:Api::Type::Array
+    item_type: Api::Type::String
+    name: "externalLoadBalancerIpv4AddressPools"
+    description: |
+      Address pools for cluster data plane external load balancing.
+  - !ruby/object:Api::Type::NestedObject
+    name: "controlPlaneEncryption"
+    description: |
+      Remote control plane disk encryption options. This field is only used when
+      enabling CMEK support.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: "kmsKey"
+        immutable: true
+        description: |
+          The Cloud KMS CryptoKey e.g.
+          projects/{project}/locations/{location}/keyRings/{keyRing}/cryptoKeys/{cryptoKey}
+          to use for protecting control plane disks. If not specified, a
+          Google-managed key will be used instead.
+      - !ruby/object:Api::Type::String
+        name: "kmsKeyActiveVersion"
+        output: true
+        description: |
+          The Cloud KMS CryptoKeyVersion currently in use for protecting control
+          plane disks. Only applicable if kms_key is set.
+      - !ruby/object:Api::Type::Enum
+        name: "kmsKeyState"
+        output: true
+        description: |
+          Availability of the Cloud KMS CryptoKey. If not `KEY_AVAILABLE`, then
+          nodes may go offline as they cannot access their local data. This can be
+          caused by a lack of permissions to use the key, or if the key is disabled
+          or deleted.
+        values:
+          - :KMS_KEY_STATE_UNSPECIFIED
+          - :KMS_KEY_STATE_KEY_AVAILABLE
+          - :KMS_KEY_STATE_KEY_UNAVAILABLE
+      - !ruby/object:Api::Type::NestedObject
+        name: "kmsStatus"
+        output: true
+        description: |
+          Error status returned by Cloud KMS when using this key. This field may be
+          populated only if `kms_key_state` is not `KMS_KEY_STATE_KEY_AVAILABLE`.
+          If populated, this field contains the error status reported by Cloud KMS.
+        properties:
+          - !ruby/object:Api::Type::Integer
+            name: 'code'
+            output: true
+            description: The status code, which should be an enum value of google.rpc.Code.
+          - !ruby/object:Api::Type::String
+            name: 'message'
+            output: true
+            description:
+              A developer-facing error message, which should be in English.
+              Any user-facing error message should be localized and sent in
+              the google.rpc.Status.details field, or localized by the client.
+  - !ruby/object:Api::Type::Enum
+    name: "status"
```

Comment:
Could `status` just be a string, since it is output-only?

Reply:
I am following the initial setup in the API proto, where the status is an enum https://source.corp.google.com/piper///depot/google3/google/cloud/edgecontainer/v1main/resources.proto;rcl=564533502;l=279

I think we might need to follow the same approach as the output from the API is enum value.

Reply:
Gotcha, thanks for the explanation. Change the Enum -> String type for the fields that is marked as output: true.